### DISCUSSION TITLE
Book chapter crossref using hyperlink worked in Bookdown but failed to render in Quarto

### DISCUSSION THREAD

**Discussion by ntluong95** [quarto-dev/quarto-cli#8878] [Q&A]
### Description

I have a bookdown project in R which has been rendered successfully. In the book, most of the time I am using conventional hyper link to refer to chapter. For example

In chapter pivoting.Rmd, the title is below without specifying the chapter label (inside the curl bracket

```
# Pivoting data {}
```
Then in chapter cleaning.Rmd, I cross reference the pivoting chapter using conventional hyperlink (without specifying the label)

```
Read more in the chapter [Pivoting data]
```

However, when I port the project to Quarto book, I have to change all .Rmd files to .qmd files, and those chapter links were broken. They are displayed as normal text. I read the [Quarto document][1], if I want to use the conventional hyperlink, either the chapter should have label or specifying the source file like this:

```
[about](about.qmd)
[about](about.qmd#section)
```

My wish is to find the way addressing the problem systematically, because I couldn't add label to chapter and then find hyperlinks to add them manually in order to proper displayed. 

  [1]: https://quarto.org/docs/books/book-crossrefs.html

**Comment by cderv** [comment]
**bookdown** works in a way that a single document is passed to Pandoc for conversion, and some feature like [`implicit_header_references`](https://pandoc.org/MANUAL.html#extension-implicit_header_references)  Pandoc's extension can be used. 

This is not really cross reference (knowing numbers of headers) but just links to headers in the document. It is mentioned as text based linked in https://bookdown.org/yihui/bookdown/cross-references.html#cross-references

Quarto works differently, and especially has a more advanced cross reference system. It is also more complex project structure and there is not a single document passed to pandoc, so feature like `implicit_header_references` only works inside a single document and not across a book HTML project. 

This is one of the differences between both tools, and indeed links need to be modified accordingly.

**Reply by ntluong95** [reply]
Thanks for flagging this. As I have many references, changing them manually seems painful. Is there anyway to do it systematically?

**Reply by mcanouil** [reply]
You could use regular expression, but in the end it won't tell in which document is located the target but only where the references are across your documents.

You could try to make a regular expression to map the reference and search for the text as headers in all documents after aggregating the results of the two searches in a file or table.
Note that because you used implicit reference, there is no guarantee there is only one header that map to one reference. Implicit does not really handle duplicated headers.
